crates/forge-runner/src/test_case_summary.rs

```diff
@@ -287,6 +291,29 @@ fn join_short_strings(data: &[Felt252]) -> String {
         .join(", ")
 }
 
+fn matches(data: &[Felt252], pattern: &[Felt252]) -> bool {
+    match (
+        convert_felts_to_byte_array_string(data),
+        convert_felts_to_byte_array_string(pattern),
+    ) {
+        // If both data and pattern are byte arrays, pattern should be a substring of data
+        (Some(data), Some(pattern)) => data.contains(&pattern),
+        // Otherwise, data should be equal to pattern
+        _ => data == pattern,
+    }
+}
+
+fn convert_felts_to_byte_array_string(data: &[Felt252]) -> Option<String> {
+    if data.first() == Some(&Felt252::try_from_hex_str(&format!("0x{BYTE_ARRAY_MAGIC}")).unwrap()) {
+        BufferReader::new(&data[1..])
+            .read::<ByteArray>()
+            .ok()
+            .map(Into::into)
+    } else {
+        None
+    }
+}
```

Comment:
We have a `ByteArray` struct in `conversions` crate, it should be used here (and if it lacks the functionality, it should be extended